src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -633,35 +643,33 @@ private List<SyntaxTrivia> GetPermutedTrivia(CSharpSyntaxNode node, List<XmlElem
             var convertedMethodGroups = nodes
                 .WhereAsArray(
                     n =>
+                    {
+                        if (!n.IsKind(SyntaxKind.IdentifierName) ||
+                            !semanticModel.GetMemberGroup(n, cancellationToken).Any())
+                        {
+                            return false;
+                        }
+
+                        ISymbol convertedType = semanticModel.GetTypeInfo(n, cancellationToken).ConvertedType;
+
+                        if (convertedType != null)
                         {
-                            if (!n.IsKind(SyntaxKind.IdentifierName) ||
-                                !semanticModel.GetMemberGroup(n, cancellationToken).Any())
-                            {
-                                return false;
-                            }
-
-                            ISymbol convertedType = semanticModel.GetTypeInfo(n, cancellationToken).ConvertedType;
-
-                            if (convertedType != null)
-                            {
-                                convertedType = convertedType.OriginalDefinition;
-                            }
-
-                            if (convertedType != null)
-                            {
-                                convertedType = SymbolFinder.FindSourceDefinitionAsync(convertedType, document.Project.Solution, cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken) ?? convertedType;
-                            }
-
-                            return Equals(convertedType, symbol.ContainingType);
-                        })
+                            convertedType = convertedType.OriginalDefinition;
+                        }
+
+                        if (convertedType != null)
+                        {
+                            convertedType = SymbolFinder.FindSourceDefinitionAsync(convertedType, document.Project.Solution, cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken) ?? convertedType;
```

Comment:
Not your fault but the WaitAndGetResult isn't good here -- this is just a .SelectAsArray() so we could just convert this to a regular foreach loop. Good follow-up item.

Reply:
Filed https://github.com/dotnet/roslyn/issues/41783